### ISSUE TITLE
Example in Java

### ISSUE THREAD

**Issue by ngocketit** [CheckoutFinland/psp-api#43]
Hi,

I'm using Java to make a request to create a new payment with the exact sample data in the Examples. However, I got `401` error. Do you have an example code in Java? Does the order of fields in the request body (the JSON string) affect the HMAC verification?

**Comment by ngocketit** [comment]
I think I made a mistake in calculating HMAC but now I got `Bad request` error. Here is the request body (I copied from the example):
```
{
   "stamp":"29858472953",
   "reference":"9187445",
   "amount":1590,
   "currency":"EUR",
   "language":"fi",
   "orderId":null,
   "items":[
      {
         "unitPrice":1590,
         "units":1,
         "vatPercentage":24,
         "productCode":"#927502759",
         "deliveryDate":"2018-03-07",
         "description":"Cat ladder",
         "category":"Pet supplies",
         "orderId":null,
         "stamp":"29858472952",
         "reference":"9187445",
         "merchant":"375917"
      }
   ],
   "customer":{
      "email":"erja.esimerkki@example.org",
      "firstName":"Erja",
      "lastName":"Esimerkki",
      "phone":"+358501234567",
      "vatId":"FI12345671"
   },
   "deliveryAddress":{
      "streetAddress":"Eteläpuisto 2 C",
      "postalCode":"33200",
      "city":"Tampere",
      "county":"Pirkanmaa",
      "country":"FI"
   },
   "invoicingAddress":{
      "streetAddress":"Gebhardinaukio 1",
      "postalCode":"00510",
      "city":"Helsinki",
      "county":"Uusimaa",
      "country":"FI"
   },
   "redirectUrls":{
      "success":"https://ecom.example.org/success",
      "cancel":"https://ecom.example.org/cancel"
   },
   "callbackUrls":{
      "success":"https://ecom.example.org/success",
      "cancel":"https://ecom.example.org/cancel"
   },
   "callbackDelay":0
}
```

**Comment by ngocketit** [comment]
There are a bunch of different error messages that are visible when I turned on debug logs. For example:
` http-outgoing-0 << "{"status":"error","message":"Request payload schema validation failed. Validate request against the provided schema.","meta":["instance.orderId is not of a type(s) string","instance.items[0] additionalProperty \"orderId\" exists in instance when not allowed"]}"`

I changed the sample data and now it works.
